pkg/controller/provider/container/ovirt/collector.go

```diff
@@ -148,9 +148,15 @@ func (r *Collector) Version() (major, minor, build, revision string, err error)
 	minor = version[1]
 	build = version[2]
 	revision = "0"
-	if len(version) > 3 {
-		revision = strings.Split(version[3], "-")[0]
+	if strings.Contains(build, "-") {
+		build = strings.Split(version[2], "-")[0]
+		revision = strings.Split(version[2], "-")[1]
+	} else {
+		if len(version) > 3 {
+			revision = strings.Split(version[3], "-")[0]
+		}
 	}
+
```

Comment:
could you please add unit tests for this? it will also serve as documentation which was missing for this part in the commit message